File: software/examples/c/example_edge_count.c
#define IPCON_EXPOSE_MILLISLEEP

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_industrial_digital_in_4_v2.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change XYZ to the UID of your Industrial Digital In 4 Bricklet 2.0

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	IndustrialDigitalIn4V2 idi4;
	industrial_digital_in_4_v2_create(&idi4, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Configure rising edge count (channel 3) with 10ms debounce
	industrial_digital_in_4_v2_set_edge_count_configuration(&idi4, 3, 0, 10);

	// Get edge count 10 times with 1s delay
	int i;
	for(i = 0; i < 10; ++i) {
		millisleep(1000);

		// Get current edge count
		uint32_t count;
		if(industrial_digital_in_4_v2_get_edge_count(&idi4, 3, false, &count) < 0) {
			fprintf(stderr, "Could not get edge count, probably timeout\n");
			return 1;
		}

		printf("Count: %u\n", count);
	}

	printf("Press key to exit\n");
	getchar();
	industrial_digital_in_4_v2_destroy(&idi4);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/go/example_edge_count.go
package main

import (
	"fmt"
	"time"
	"tinkerforge/industrial_digital_in_4_v2_bricklet"
	"tinkerforge/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Industrial Digital In 4 Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	idi4, _ := industrial_digital_in_4_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Configure rising edge count (channel 3) with 10ms debounce
	idi4.SetEdgeCountConfiguration(3, 0, 10)

	// Get edge count 10 times with 1s delay
	for i := 0; i < 10; i++ {
		time.Sleep(1000 * time.Millisecond)

		// Get current edge count.
		count, _ := idi4.GetEdgeCount(3, false)
		fmt.Printf("Count: \n", count)
	}

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
